android/src/main/java/com/swmansion/reanimated/nodes/ConcatNode.java

```diff
@@ -3,21 +3,36 @@
 import com.facebook.react.bridge.ReadableMap;
 import com.swmansion.reanimated.NodesManager;
 import com.swmansion.reanimated.Utils;
+import java.text.NumberFormat;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.Locale;
 
 public class ConcatNode extends Node {
+  private final DecimalFormatSymbols symbols;
+  private final NumberFormat nf;
+
   private final int[] mInputIDs;
 
   public ConcatNode(int nodeID, ReadableMap config, NodesManager nodesManager) {
     super(nodeID, config, nodesManager);
     mInputIDs = Utils.processIntArray(config.getArray("input"));
+    symbols = new DecimalFormatSymbols(Locale.getDefault());
+    symbols.setDecimalSeparator('.');
+    nf = new DecimalFormat("##.###", symbols);
```

Comment:
Can we leave this change out of this PR. It does not seem relevant IMO

Reply:
Sure. For some context, the original point of the PR was to get string animated values working on Android. And without this change almost all the use cases for string animated Values actually break because the animated Double values display always with a decimal point even when they are whole integers (eg 12.0 rather than 12). Without this `concat(new Value(12))` will return different results on iOS and Android - so should I rather put this particular change it in another PR then?

Reply:
@kmagiera please confirm if I should still move this to a separate PR?